src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,169 @@
-
+        if (size >= threshold) {
+            threshold *= DOUBLING_FACTOR;
+            resize();
+        }
+        Node<K, V> newNode = new Node<>(hash(key), key, value, null);
+        int index = (table.length - 1) & newNode.hash;
+        if (table[index] == null) {
+            justAddNewNode(index, newNode);
+        } else {
+            Node<K, V> current = table[index];
+            while (current.next != null) {
+                if (theseKeysAreEqual(current.key, newNode.key)) {
+                    reWrightValue(current, newNode);
+                    return;
+                } else {
+                    current = current.next;
+                }
+            }
+            if (theseKeysAreEqual(current.key, newNode.key)) {
+                reWrightValue(current, newNode);
+                return;
+            }
+            addNextNewNode(current, newNode);
+        }
     }
 
     @Override
     public V getValue(K key) {
+        int index = (table.length - 1) & hash(key);
+        if (index < table.length && table[index] != null) {
+            Node<K, V> current = table[index];
+            while (current != null) {
+                if (current.hash == hash(key)
+                        && Objects.equals(current.key, key)) {
+                    return current.value;
+                } else {
+                    current = current.next;
+                }
+            }
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    @Override
+    public String toString() {
+        return "table=" + Arrays.toString(table);
+    }
+
+    @SuppressWarnings({"unchecked"})
+    private void resize() {
+        Node<K,V>[] oldTable = table;
+        table = (Node<K,V>[]) new Node[oldTable.length * DOUBLING_FACTOR];
+        size = 0;
+        transfer(oldTable);
+    }
+
+    private void justAddNewNode(int index, Node<K, V> newNode) {
+        table[index] = newNode;
+        size++;
+    }
+
+    private void addNextNewNode(Node<K, V> current, Node<K, V> newNode) {
+        current.next = newNode;
+        size++;
+    }
+
+    private boolean theseKeysAreEqual(K currentKey, K newKey) {
+        return hash(currentKey) == hash(newKey)
+                && (Objects.equals(currentKey, newKey));
+    }
+
+    private void reWrightValue(Node<K, V> current, Node<K, V> newNode) {
+        current.value = newNode.value;
+    }
+
+    private void transfer(Node<K, V>[] oldTable) {
+        for (Node<K, V> oldNode : oldTable) {
+            if (oldNode != null) {
+                Node<K, V> current = oldNode;
+                while (current != null) {
+                    put(current.key, current.value);
+                    current = current.next;
+                }
+            }
+        }
+    }
+
+    private int hash(Object key) {
+        return (key == null) ? 0 : key.hashCode();
+    }
+
+    private static class Node<K, V> {
```

Comment:
Don't use getters and setters in the Node class. Since we have access to private fields of the inner class, using getters or setters is redundant. Directly accessing the fields would be more appropriate.